humphd/my-photohub
# Initial implementation for #1

## Description
This PR implements the features requested in #1. 

## Types of Changes


## Checklist

- [x] Authenticate a GitHub user using Personal Authentication Token (PAT) using `octokit`
- [x] Create a public repo on behalf of the authenticated user using `octokit`

## Additional informationn
- This PR is to get initial feedback on the implementation from the maintainers.
-  I opted to use PAT to authenticate GitHub users as it's safer than user entering their GitHub username and password directly into the form fields. We can always switch to authenticating users using the username and password and keeping PAT as an optional feature. 
- User can create more than one repo. If needed, we can change this behaviour later by checking if they have an existing repo generated using `my-photohub`.

## Steps to Test
1. Navigate to GitHub and [create a PAT token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic). Make sure you checkmark `repos` from the list of permissions before generating it.
2. Install the `dependencies` using `npm install`.
3. Start `my-photohub` by running `node src/api/index.js` from the terminal
4. Use a browser to navigate `localhost:8080`.
5. Enter the PAT key in the form field and click on `Authenticate`
6. Valid PAT key will show a OK status message in the browser window.  Bad input will show an error message. Non exisiting PAT key will show Unauthorized message.
7. After seeing an OK message on `my-photohub`, go to your GitHub account to see a repo was generated on your behalf:
<img width="956" alt="Screenshot 2022-11-03 at 10 31 39 PM" src="https://user-images.githubusercontent.com/50856799/199872729-25cb5d95-a90c-4579-ba2b-190f8fa6f844.png">
